test/Microsoft.AspNet.Mvc.Core.Test/DefaultAssemblyProviderTests.cs

```diff
@@ -164,5 +205,100 @@ public NullAssemblyProvider(ILibraryManager libraryManager) : base(libraryManage
             {
             }
         }
+
+        private class MvcAssembliesTestingProvider : DefaultAssemblyProvider
+        {
+            private static readonly ILibraryManager _libraryManager = GetLibraryManager();
+            private static readonly string _mvcName = "Microsoft.AspNet.Mvc";
+            private static HashSet<string> _mvcReferenceAssemblies;
+
+            public MvcAssembliesTestingProvider() : base(_libraryManager)
+            { }
+
+            public HashSet<string> LoadableReferenceAssemblies
+            {
+                get
+                {
+                    if (_mvcReferenceAssemblies != null)
+                    {
+                        return _mvcReferenceAssemblies;
+                    }
+
+                    var dependencies = new HashSet<string>() { _mvcName };
+                    dependencies = GetAllDepedencies(_mvcName, dependencies);
+
+                    _mvcReferenceAssemblies = new HashSet<string>(
+                        SelectMvcAssemblies(
+                            GetLoadableAssemblies(dependencies)));
+
+                    return _mvcReferenceAssemblies;
+                }
+            }
+
+            public new HashSet<string> ReferenceAssemblies
+            {
+                get
+                {
+                    return base.ReferenceAssemblies;
+                }
+            }
+
+            private static HashSet<string> GetAllDepedencies(string libraryName, HashSet<string> dependencies)
+            {
+                var directDependencies = _libraryManager.GetLibraryInformation(libraryName).Dependencies;
+
+                if (directDependencies != null && directDependencies.Count() > 0)
```

Comment:
`Count() > 0` is not needed. This will do the same thing without it. Think before you `Count()` -- its almost always a code smell.